mlrun/feature_store/retrieval/base.py

```diff
@@ -165,8 +169,163 @@ def _generate_vector(
+            node.data["save_cols"] += node.data["save_index"]
+            if feature_set.spec.timestamp_key:
+                entity_timestamp_column_list = [feature_set.spec.timestamp_key]
+                column_names += entity_timestamp_column_list
+                node.data["save_cols"] += entity_timestamp_column_list
+                if not entity_timestamp_column:
+                    # if not entity_timestamp_column the firs `FeatureSet` will define it
+                    entity_timestamp_column = feature_set.spec.timestamp_key
+
+            # rename columns to be unique for each feature set and select if needed
+            rename_col_dict = {
+                column: f"{column}_{name}"
+                for column in column_names
+                if column not in node.data["save_cols"]
+            }
+            fs_entities = list(feature_set.spec.entities.keys())
+            df_temp = self._rename_columns_and_select(
+                df, rename_col_dict, columns=list(set(column_names + fs_entities))
+            )
+
+            df = df_temp if df_temp is not None else df
+
+            dfs.append(df)
+            del df
+            del df_temp
+
+            keys.append([node.data["left_keys"], node.data["right_keys"]])
+
+            # update alias according to the unique column name
+            new_columns = []
+            if not self._drop_indexes:
+                new_columns.extend([(ind, ind) for ind in fs_entities])
+            for column, alias in columns:
+                if column in rename_col_dict and alias:
+                    new_columns.append((rename_col_dict[column], alias))
+                elif column in rename_col_dict and not alias:
+                    new_columns.append((rename_col_dict[column], column))
+                else:
+                    new_columns.append((column, alias))
+            self._update_alias(
+                dictionary={name: alias for name, alias in new_columns if alias}
+            )
+
+        # convert pandas entity_rows to spark DF if needed
+        if (
+            entity_rows is not None
+            and not hasattr(entity_rows, "rdd")
+            and self.engine == "spark"
+        ):
+            entity_rows = self.spark.createDataFrame(entity_rows)
+
+        # join the feature data frames
+        self.merge(
+            entity_df=entity_rows,
+            entity_timestamp_column=entity_timestamp_column,
+            featuresets=feature_sets,
+            featureset_dfs=dfs,
+            keys=keys,
+        )
+
+        all_columns = None
+        if not self._drop_indexes and entity_timestamp_column:
+            if entity_timestamp_column not in self._alias.values():
+                self._update_alias(
+                    key=entity_timestamp_column, val=entity_timestamp_column
+                )
+                all_columns = list(
+                    set([entity_timestamp_column] + list(self._alias.keys()))
+                )
```

Comment:
Why do you need to add `[entity_timestamp_column]` here? If you did `_update_alias`, won't it be included in `self._alias.keys()`?